src/DataStructures/VectorImpl.hpp

```diff
@@ -199,6 +199,18 @@ class VectorImpl
     std::fill(data(), data() + set_size, value);
   }
 
+  // Create from a copy of the given container
+  ///
+  /// \param container A container with a `value_type` that is the same as `T`,
+  /// e.g. `std::vector<T>` or `std::array<T>`.
+  template <typename Container, typename U = typename Container::value_type,
+            Requires<std::is_same_v<U, T>> = nullptr>
+  explicit VectorImpl(const Container& container)
```

Comment:
this is extremely generic and I'm worried people will do silly things that will be bad for performance. Can you please restrict it to vector & array explicitly?